dds/DCPS/XTypes/DynamicData.cpp

```diff
@@ -0,0 +1,1068 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include "DynamicData.h"
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace XTypes {
+
+DDS::ReturnCode_t DynamicData::get_descriptor(MemberDescriptor& value, MemberId id) const
+{
+}
+
+DDS::ReturnCode_t DynamicData::set_descriptor(MemberId id, const MemberDescriptor& value)
+{
+}
+
+bool DynamicData::equals(const DynamicData& other) const
+{
+}
+
+MemberId DynamicData::get_member_id_by_name(String name) const
+{
+}
+
+MemberId DynamicData::get_member_id_at_index(ACE_CDR::ULong) const
+{
+}
+
+ACE_CDR::ULong DynamicData::get_item_count() const
+{
+}
+
+DDS::ReturnCode_t DynamicData::clear_all_values()
+{
+}
+
+DDS::ReturnCode_t DynamicData::clear_nonkey_values()
+{
+}
+
+DDS::ReturnCode_t DynamicData::clear_value(MemberId id)
+{
+}
+
+DynamicData DynamicData::loan_value(MemberId id)
+{
+}
+
+DDS::ReturnCode_t DynamicData::return_loaned_value(const DynamicData& value)
+{
+}
+
+DynamicData DynamicData::clone() const
+{
+}
+
+DDS::ReturnCode_t DynamicData::get_int32_value(ACE_CDR::Long& value, MemberId id) const
+{
+  const OPENDDS_MAP<MemberId, size_t>::const_iterator it = offset_lookup_table_.find(id);
+  if (it != offset_lookup_table_.cend()) {
+    size_t offset = it->second;
+    std::ostringstream err_msg;
+    err_msg << "Failed to skip to the member with ID " << id;
```

Comment:
This looks pretty costly, use a ostreamstream to construct an error string to pass to skip, where it is only used when there is an error and debug level >= 10